htdocs/core/boxes/box_accountancy_unbalanced_entries.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/* Copyright (C) 2003-2007	Rodolphe Quiedeville	<rodolphe@quiedeville.org>
+ * Copyright (C) 2004-2009	Laurent Destailleur		<eldy@users.sourceforge.net>
+ * Copyright (C) 2005-2009	Regis Houssin			<regis.houssin@inodbox.com>
+ * Copyright (C) 2022		Alexandre Spangaro		<aspangaro@open-dsi.fr>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ *		\file       htdocs/core/boxes/box_accountancy_unbalanced_entries.php
+ *		\ingroup    Accountancy
+ *		\brief      Module to generated widget of unbalanced entries
+ */
+
+include_once DOL_DOCUMENT_ROOT.'/core/boxes/modules_boxes.php';
+
+
+/**
+ * Class to manage the box to show unbalanced entries
+ */
+class box_accountancy_unbalanced_entries extends ModeleBoxes
+{
+	public $boxcode = "accountancy_unbalanced_entries";
+	public $boximg = "accounting";
+	public $boxlabel = "BoxUnbalancedEntries";
+	public $depends = array("accounting");
+
+	/**
+	 * @var DoliDB Database handler.
+	 */
+	public $db;
+
+	public $param;
+
+	public $info_box_head = array();
+	public $info_box_contents = array();
+
+
+	/**
+	 *  Constructor
+	 *
+	 *  @param  DoliDB  $db         Database handler
+	 *  @param  string  $param      More parameters
+	 */
+	public function __construct($db, $param)
+	{
+		global $user;
+
+		$this->db = $db;
+
+		$this->hidden = !($user->rights->accounting->mouvements->lire);
+	}
+
+	/**
+	 *  Load data for box to show them later
+	 *
+	 *  @param	int		$max        Maximum number of records to load
+	 *  @return	void
+	 */
+	public function loadBox($max = 5)
+	{
+		global $user, $langs, $conf;
+
+		include_once DOL_DOCUMENT_ROOT.'/accountancy/class/bookkeeping.class.php';
+
+		$bookkeepingstatic = new BookKeeping($this->db);
+
+		$this->info_box_head = array('text' => $langs->trans("BoxTitleUnbalancedEntries", $max));
+
+		if ($user->rights->accounting->mouvements->lire) {
+			$sql = "SELECT DISTINCT b.piece_num";
```

Comment:
Seems not valid sql.
You have a group by that does not contains all fields in the select. Also there is no count or min or max function.
I think you want the SUM(b.debit) and the SUM(b.credit) into the select too ?

Reply:
@eldy Thanks for your return.

Yes, we seems to have a problem with databases who have a "only full group by" option activated but when I try to list all field in group by, my request doesn't work.

What I want is to have a list of unbalanced accounting entries. Only one line with the piece_num as long as the sum of the debits and credits of the accounting entry are not equal.

Normally, it is not possible to have unbalanced entries in Dolibarr but on imports or advanced manipulations, we can meet unbalances so it will allow to have a widget that will detect the errors to help to solve them.

Reply:
So this is what you must do to get only 1 line:
remove the distint on the b.piece_num,
remove the line 
$sql .= ", b.doc_date as date_movement";
$sql .= ", b.label_operation";
$sql .= ", b.montant as amount";
$sql .= ", b.code_journal";
and add instead add
$sql .= " SUM(b.debit), SUM(b.credit)"
You should get 1 line for each piece_num with the debit and credit on same line, and only when debit != credit.